sources/ghostdog/src/main.rs

```diff
@@ -319,73 +372,43 @@ lazy_static! {
     ].iter().copied().collect();
 }
 
-// Returning a Result from main makes it print a Debug representation of the error, but with Snafu
-// we have nice Display representations of the error, so we wrap "main" (run) and print any error.
-// https://github.com/shepmaster/snafu/issues/110
-fn main() {
-    if let Err(e) = run() {
-        eprintln!("{}", e);
-        std::process::exit(1);
-    }
-}
+/// write_config_string will write the provided string to the provided path
+fn write_config_string(config_path: PathBuf, content_string: &str) -> Result<()> {
+    // Create a temporary file in the desired config directory
+    // Check that a path was specified and isn't /
+    let env_directory = config_path
+        .parent()
+        .filter(|p| !p.as_os_str().is_empty() && *p != Path::new("/"))
+        .ok_or_else(|| error::Error::NoParentDirectory {
+            path: config_path.clone(),
+        })?;
+
+    let mut tempfile =
+        NamedTempFile::new_in(env_directory).context(error::CreateTempFileSnafu {
+            path: PathBuf::from(env_directory),
+        })?;
+
+    // Write the config to the temporary file
+    writeln!(tempfile, "{}", content_string).context(error::WriteTempFileSnafu)?;
```

Comment:
nit: avoid the `writeln!` if the content string is empty so the file is zero bytes